haystack/core/pipeline/pipeline.py

```diff
@@ -40,6 +40,72 @@ class Pipeline(PipelineBase):
     Orchestrates component execution according to the execution graph, one after the other.
     """
 
+    def _warn_if_ambiguous_intent(
```

Comment:
@Amnah199 I started out with this method to check for ambiguous execution order up front (e.g. the case that I showed for the cycle with 2 prompt builders in the PR description).

The problem is, that we don't know for sure that the running order will be ambiguous, we just know that it's possible.

I then thought that we might just check this when actually running the pipeline and only warn if this case occurs (see comment further down).

I think that would be a better approach and I'd skip this one then. What do you think?